.ci/packer_cache.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+SCRIPT="$0"
+
+# SCRIPT might be an arbitrarily deep series of symbolic links; loop until we
+# have the concrete path
+while [ -h "$SCRIPT" ] ; do
+  ls=$(ls -ld "$SCRIPT")
+  # Drop everything prior to ->
+  link=$(expr "$ls" : '.*-> \(.*\)$')
+  if expr "$link" : '/.*' > /dev/null; then
+    SCRIPT="$link"
+  else
+    SCRIPT=$(dirname "$SCRIPT")/"$link"
+  fi
+done
+
+source $(dirname "${SCRIPT}")/java-versions.properties
+JAVA_HOME="${HOME}"/.java/${ES_BUILD_JAVA} ./gradlew resolveAllDependencies --parallel
```

Comment:
I don't believe the `$ES_BUILD_JAVA` environment variable will be available when this runs. It's triggered as part of a job within infra-ci that doesn't have that as a parameter, and is run via an Ansible task [1] which again clears then environment unless it's explicitly defined. Would hard-coding this to `"${HOME}/.java/java8` work for this case? Note that at the moment, the caches will build from whatever is put into the elasticsearch repo's _master_ branch, but are open to expanding functionality in the future if need be.

[1] https://github.com/elastic/infra/blob/master/ansible/roles/packer_cache/tasks/linux/packer_cache.yml

Reply:
Oh, never mind...you're sourcing `java-versions.properties` yourself.

Reply:
Exactly. 😇

Reply:
You might want to make this script executable in the repo. I believe we explicitly call it with bash to prevent problems, but it would be good to have it work either way.

Reply:
That's interesting, I thought that I did, let me check.

Reply:
@elasticdog So I think that I did, are you seeing something different?

```
14:50:57 1d [jason@totoro:/tmp] $ mkdir -p jasontedor
14:51:03 1d [jason@totoro:/tmp] $ cd jasontedor/
14:51:05 1d [jason@totoro:/tmp/jasontedor] $ git clone -o jasontedor git@github.com:jasontedor/elasticsearch.git && git checkout cache-dependencies
Cloning into 'elasticsearch'...
remote: Counting objects: 1232571, done.
remote: Compressing objects: 100% (334/334), done.
remote: Total 1232571 (delta 250), reused 34 (delta 27), pack-reused 1232210
Receiving objects: 100% (1232571/1232571), 468.71 MiB | 34.14 MiB/s, done.
Resolving deltas: 100% (669388/669388), done.
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
14:51:48 1d [jason@totoro:/tmp/jasontedor] 128 $ cd elasticsearch/
14:52:02 1d [jason@totoro:/tmp/jasontedor/elasticsearch] master ± git checkout cache-dependencies
Branch 'cache-dependencies' set up to track remote branch 'cache-dependencies' from 'jasontedor'.
Switched to a new branch 'cache-dependencies'
14:52:06 1d [jason@totoro:/tmp/jasontedor/elasticsearch] cache-dependencies ± ls -al .ci/packer_cache.sh 
-rwxrwxr-x. 1 jason jason 515 Sep 14 14:52 .ci/packer_cache.sh
14:52:10 1d [jason@totoro:/tmp/jasontedor/elasticsearch] cache-dependencies ± 
```

Reply:
You are correct...I'm used to seeing the GitHub UI dump it in the PR, but maybe that's only on permission changes.

Reply:
Cool, thanks for checking.